lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -121,15 +121,65 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     epoch * slots_per_epoch
   end
 
+  @doc """
+  Return from ``indices`` a random index sampled by effective balance.
+  """
+  @spec compute_proposer_index(BeaconState.t(), [SszTypes.validator_index()], SszTypes.bytes32()) ::
+          SszTypes.validator_index() | {:error, binary()}
+  def compute_proposer_index(state, indices, seed) do
+    if not (length(indices) > 0) do
+      {:error, "Empty indices"}
+    else
+      compute_proposer_index(state, indices, seed, 0)
```

Comment:
This should be wrapped in `{:ok, ...}` for less error-prone matching:

```elixir
# we can do
case compute_proposer_index(...) do
  {:ok, index} -> ...
  {:error, reason} -> ...
end

# instead of
case compute_proposer_index(...) do
  index -> ...     # this is WRONG
  {:error, reason} -> ...
end
```